### PR TITLE
Handle special characters in TraceProxyRequestHelper

### PR DESCRIPTION
**Pull request by Qerny**

### CODE REVIEW

**Comment by spencergibb** [comment]
Please add a test that fails prior to this change and succeds with it

**Comment by Qerny** [comment]
> Please add a test that fails prior to this change and succeds with it

Done, check, please

**Comment by OlgaMaciaszek** [comment]
Fixes gh-3873.

File: spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		@Override
+		public String getMethod() {
+			return request.getMethod();
+		}
+
+		@Override
+		public URI getUri() {
+			String queryString = this.request.getQueryString();
+			if (!StringUtils.hasText(queryString)) {
+				return URI.create(this.request.getRequestURL().toString());
```
Comment (OlgaMaciaszek) [inline spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequest.java:54]: Please remove unnecessary use of `this.` throughout this class.

File: spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.boot.actuate.trace.http.TraceableRequest;
+import org.springframework.util.StringUtils;
+import org.springframework.web.util.UriUtils;
+
+/**
+ * @author Spencer Gibb
+ */
```
Comment (OlgaMaciaszek) [inline spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequest.java:36]: Please add your full name with `@author` tag to all modified classes.

File: spring-cloud-netflix-zuul/src/test/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequestTests.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import org.junit.Test;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link ServletTraceableRequest}.
+ */
+public class ServletTraceableRequestTests {
```
Comment (OlgaMaciaszek) [inline spring-cloud-netflix-zuul/src/test/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequestTests.java:29]: Please use jUnit Jupiter in tests and then change class and methods access to package-private.

File: spring-cloud-netflix-zuul/src/test/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequestTests.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+	private MockHttpServletRequest request;
+
+	@Before
+	public void setup() {
+		this.request = new MockHttpServletRequest("GET", "/script");
```
Comment (OlgaMaciaszek) [inline spring-cloud-netflix-zuul/src/test/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequestTests.java:35]: Please remove unnecessary use of `this` keyword throughout this class.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
@Qerny Thanks for submitting the PR. The logic looks good, have added comments about some minor, cosmetic issues - please address and this will be good to merge.

File: spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequest.java (hunk 1 above)
Comment (Qerny) [reply spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequest.java]: Done

File: spring-cloud-netflix-zuul/src/test/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequestTests.java (hunk 3 above)
Comment (Qerny) [reply spring-cloud-netflix-zuul/src/test/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequestTests.java]: Done

File: spring-cloud-netflix-zuul/src/test/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequestTests.java (hunk 4 above)
Comment (Qerny) [reply spring-cloud-netflix-zuul/src/test/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequestTests.java]: Done

Comment (Qerny) [reply spring-cloud-netflix-zuul/src/test/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequestTests.java]: Done

File: spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequest.java (hunk 2 above)
Comment (Qerny) [reply spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/ServletTraceableRequest.java:36]: Done

**Review by OlgaMaciaszek — APPROVED** [review]
LGTM.

**Pull request merged**